pageserver/src/tenant.rs

```diff
@@ -3405,30 +3410,42 @@ fn run_initdb(
         initdb_bin_path, initdb_target_dir, initdb_lib_dir,
     );
 
-    let initdb_output = Command::new(&initdb_bin_path)
+    let _permit = INIT_DB_SEMAPHORE.acquire().await;
+
+    let initdb_command = tokio::process::Command::new(&initdb_bin_path)
         .args(["-D", initdb_target_dir.as_ref()])
         .args(["-U", &conf.superuser])
         .args(["-E", "utf8"])
         .arg("--no-instructions")
-        // This is only used for a temporary installation that is deleted shortly after,
-        // so no need to fsync it
         .arg("--no-sync")
         .env_clear()
         .env("LD_LIBRARY_PATH", &initdb_lib_dir)
         .env("DYLD_LIBRARY_PATH", &initdb_lib_dir)
         .stdout(Stdio::null())
-        .output()
+        .spawn()
         .with_context(|| {
             format!(
                 "failed to execute {} at target dir {}",
                 initdb_bin_path, initdb_target_dir,
             )
         })?;
-    if !initdb_output.status.success() {
-        bail!(
-            "initdb failed: '{}'",
-            String::from_utf8_lossy(&initdb_output.stderr)
-        );
+
+    let child_id = initdb_command.id();
+
+    tokio::select! {
+        initdb_output = initdb_command.wait_with_output() => {
```

Comment:
This method has no cancellation safety documentation,. We should do the work (check more recent tokio upstream version first!) to document this.

Reply:
It takes ownership so I'd call it cancellation safe, by the implementation it also looks that way.

We just need the `kill_on_drop` configured before we start racing with the cancellation.